src/components/AddBookButton.js

```diff
@@ -1,21 +1,23 @@
 import React from 'react';
+import { v4 as uuidv4 } from 'uuid';
 import { useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
-import { addBook } from '../redux/books/booksSlice';
+import { addBook } from '../api/middleware';
 
-const AddBookButton = ({ bookTitle, bookAuthor }) => {
+const AddBookButton = ({ bookTitle, bookAuthor, bookCategory }) => {
   const dispatch = useDispatch();
 
   const handleAddBook = () => {
-    const id = Math.floor(Math.random() * 100);
     const newBook = {
-      item_id: id,
+      item_id: uuidv4(),
       title: bookTitle,
       author: bookAuthor,
+      category: bookCategory,
     };
 
     dispatch(addBook(newBook));
   };
```

Comment:
- [ ] Kindly fix the following behavior. Currently, this is what happens when the user tries to add or remove a book. (See attached GIF)

![chrome-capture-2023-6-26](https://github.com/RileyManda/bookstore/assets/62290545/350a50c6-1692-478e-93fc-4f2135a45555)

Reply:
Thank you for reviewing my implementation and providing feedback. Looks like I might have broken the dispatch on add and remove functionality and will work on the changes. Thank you